test/e2e/receive_test.go

```diff
@@ -101,51 +187,161 @@ func TestReceive(t *testing.T) {
 
 		testutil.Ok(t, q.WaitSumMetricsWithOptions(e2e.Equals(3), []string{"thanos_store_nodes_grpc_connections"}, e2e.WaitMissingMetrics))
 
-		queryAndAssertSeries(t, ctx, q.HTTPEndpoint(), queryUpWithoutInstance, promclient.QueryOptions{
+		result := instantQuery(t, ctx, q.HTTPEndpoint(), "count(up) by (prometheus)", promclient.QueryOptions{
 			Deduplicate: false,
-		}, []model.Metric{
-			{
-				"job":        "myself",
-				"prometheus": "prom1",
-				"receive":    "i2",
-				"replica":    "0",
-				"tenant_id":  "default-tenant",
-			},
-			{
-				"job":        "myself",
-				"prometheus": "prom2",
-				"receive":    "i1",
-				"replica":    "0",
-				"tenant_id":  "default-tenant",
+		}, 3)
+
+		// TODO(ianbillett): change this assertion to testutil.Equals when #4359 is fixed.
+		// Currently, we expect this e2e test to fail due to bugs in the reicever split proposal implementation.
+		for _, series := range result {
+			testutil.NotEqual(t, series.Value, 2.0)
```

Comment:
Why not equal something else with TODO?

Reply:
Essentially this can be surprising. So easy to match this assertion (so many options that passes). It essentially does not give any value